src/DotNetWorker/Diagnostics/FunctionInvocationScope.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Functions.Worker.Diagnostics
+{
+    internal class FunctionInvocationScope : IReadOnlyList<KeyValuePair<string, object>>
```

Comment:
I took this design from here: https://github.com/grpc/grpc-dotnet/blob/ec6de5c047/src/Grpc.Net.Client/Internal/GrpcCallScope.cs